File: 4. Absolute Difference of 1/main.cpp
class Solution 
{
  public:
    vector<int> getDigitDiff1AndLessK(vector<int>& arr, int k) 
    {
        // code here
        int size = 0;
        
        for (int i: arr) 
        {
            if (i < k && i > 9) 
            {
                if (isAdjacent(i)) 
                {
                    arr[size] = i;
                    size++;
                }
            }
        }
        
        arr.resize(size);
        return arr;
    }
    
    bool isAdjacent(int n) 
    {
        bool flag = true;
        
        while (flag && n > 9) 
        {
            int d1 = n % 10;
            n = n / 10;
            int d2 = n % 10;
            
            flag = (d1 + 1 == d2) || (d1 - 1 == d2);
        }
        
        return flag;
    }
};

File: 4. Absolute Difference of 1/main.py
class Solution:
    def getDigitDiff1AndLessK(self, arr, k):
        # code here
        size = 0
        
        for i in arr:
            if i < k and i > 9:
                if self.isAdjacent(i):
                    arr[size] = i
                    size += 1
                
        return arr[0:size]
    
    def isAdjacent(self, n):
        flag = True
        
        while flag and n > 9:
            d1 = n % 10
            n = n // 10
            d2 = n % 10
            
            flag = (d1 + 1 == d2) or (d1 - 1 == d2)            
        
        return flag